metrics/ci_integrations/test/firestore/adapter/firestore_storage_client_adapter_test.dart

```diff
@@ -0,0 +1,176 @@
+// ignore_for_file: avoid_implementing_value_types
+
+import 'package:ci_integration/firestore/adapter/firestore_storage_client_adapter.dart';
+import 'package:ci_integration/firestore/deserializer/build_data_deserializer.dart';
+import 'package:firedart/firedart.dart';
+import 'package:metrics_core/metrics_core.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+import 'package:grpc/grpc.dart';
+
+void main() {
+  group("FirestoreStorageClientAdapter", () {
+    const testProjectId = 'projectId';
+    const testDocumentId = 'documentId';
+    final buildDataTestJson = {
+      'buildNumber': 1,
+      'startedAt': DateTime.now(),
+      'buildStatus': BuildStatus.failed.toString(),
+      'duration': 100,
+      'workflowName': 'testWorkflowName',
+      'url': 'testUrl',
+      'coverage': 0.1,
+    };
+    final _firestoreMock = _FirestoreMock();
+    final _collectionReferenceMock = _CollectionReferenceMock();
+    final _documentReferenceMock = _DocumentReferenceMock();
+    final _documentMock = _DocumentMock();
+    final mockAdapter = FirestoreStorageClientAdapter(_firestoreMock);
+
+    PostExpectation<Future<Document>> whenFetchProject({
+      String collectionId = 'projects',
+      String projectId = testProjectId,
+    }) {
+      when(_firestoreMock.collection(collectionId))
+          .thenReturn(_collectionReferenceMock);
+      when(_collectionReferenceMock.document(projectId))
+          .thenReturn(_documentReferenceMock);
+      return when(_documentReferenceMock.get());
+    }
+
+    PostExpectation<Future<List<Document>>> whenFetchLastBuild({
+      String collectionId = 'build',
+      String whereFieldPath = 'projectId',
+      dynamic isEqualTo = testProjectId,
+      String orderByFieldPath = 'startedAt',
+      int limit = 1,
+    }) {
+      when(_firestoreMock.collection(collectionId))
+          .thenReturn(_collectionReferenceMock);
+      when(_collectionReferenceMock.where(whereFieldPath, isEqualTo: isEqualTo))
+          .thenReturn(_collectionReferenceMock);
+      when(_collectionReferenceMock.orderBy(orderByFieldPath, descending: true))
+          .thenReturn(_collectionReferenceMock);
+      when(_collectionReferenceMock.limit(limit))
+          .thenReturn(_collectionReferenceMock);
+      return when(_collectionReferenceMock.getDocuments());
+    }
+
+    setUp(() {
+      reset(_firestoreMock);
+      reset(_collectionReferenceMock);
+      reset(_documentReferenceMock);
+      reset(_documentMock);
+    });
+
+    test(
+      "should throw ArgumentError trying to create an instance with null Firestore",
+      () {
+        expect(() => FirestoreStorageClientAdapter(null), throwsArgumentError);
+      },
+    );
+
+    test(
+      ".addBuilds() should return normally if firestore throws GrpcError.notFound",
+      () {
+        whenFetchProject().thenThrow(GrpcError.notFound());
+        expect(() => mockAdapter.addBuilds(testProjectId, []), returnsNormally);
```

Comment:
Given that the result of calling the `addBuilds` function is a future, I'd rather recommend using test API for futures. Thus, 
```suggestion
        final result = mockAdapter.addBuilds(testProjectId, []);
        
        expect(result , completes);
```
What do you think?